google/cloud/firestore_v1/_helpers.py

```diff
@@ -219,6 +232,36 @@ def encode_dict(values_dict) -> dict:
     return {key: encode_value(value) for key, value in values_dict.items()}
 
 
+def document_snapshot_to_protobuf(snapshot: "google.cloud.firestore_v1.base_document.DocumentSnapshot") -> Optional["google.cloud.firestore_v1.types.Document"]:  # type: ignore
```

Comment:
Could this be added as an internal method on a DocumentSnapshot? ie `DocumentSnapshot._to_protobuf()`

Reply:
It sure could. We may have miscommunicated, but I had understood one of our exchanges to mean this was the desired location 😆

Reply:
PS - It is exposed in that way - the logic is just here (which is what I had understood us to settle on originally). Curious to hear whether that's the extent of what you meant, or if you're thinking of moving this entire block into that method.

Reply:
I unfortunately do not recall the conversation :( I just found it unusual and worth asking :)

Reply:
Resolving this since the class's `_to_protobuf()` exists and calls this method (which is a pattern that exists elsewhere in the library).